src/pfe/file-watcher/server/src/projects/projectSpecifications.ts

```diff
@@ -453,6 +453,81 @@ const reconfigIgnoredFilesForDaemon = async function (ignoredPaths: string[], op
     io.emitOnListener("projectSettingsChanged", data);
 };
 
+/**
+ * @function
+ * @description Reconfig the ref paths for a project.
+ *
+ * @param args <Required | Any> - Required arguments for reconfiguration of ref paths.
+ *
+ * @returns Promise<any>
+ */
+const reconfigRefPathsForDaemon = async function (refPaths: RefPath[], operation: Operation): Promise<any> {
+    const projectInfo: ProjectInfo = operation.projectInfo;
+    const projectID = projectInfo.projectID;
+
+    // if defined, do some checks, sanitized the input etc
+    if (refPaths) {
+        if (!(refPaths instanceof Array)) {
+            const errorMsg = "BAD_REQUEST: refPaths must be an array.";
+            logger.logProjectError("Failed to update ref path list for file watching: " + errorMsg, projectID);
+
+            const data: ProjectSettingsEvent = {
+                operationId: operation.operationId,
+                projectID: projectID,
+                refPaths: undefined,
+                status: "failed",
+                error: errorMsg
+            };
+            io.emitOnListener("projectSettingsChanged", data);
+            return;
+        }
+
+        // sanitize the input
+        const sanitized: RefPath[] = [];
+        refPaths.forEach((el) => {
+            const refPath = RefPath.createFrom(el);
+            if (refPath) {
+                sanitized.push(refPath);
+            }
+        });
+        refPaths = sanitized;
+        logger.logProjectInfo("Attempting to set the ref path list for file watching with: " + JSON.stringify(refPaths), projectID);
+
+        if (projectInfo.refPaths instanceof Array &&
+            projectInfo.refPaths.length === refPaths.length &&
+            // for every old ref saved in projectInfo, can find a new ref in refPaths where the from and to properties are equal
+            projectInfo.refPaths.every((oldRef) => {
+                return refPaths.findIndex(newRef => newRef.from === oldRef.from && newRef.to === oldRef.to) != -1;
+            })) {
+                logger.logProjectInfo("The ref path list for file watching is unchanged", projectID);
+                return;
+        }
+    }
+    // if undefined and previously it was also undefined, nothing changed
+    else if (!projectInfo.refPaths) {
+        logger.logProjectInfo("The ref path list for file watching is unchanged", projectID);
+        return;
+    }
+    // otherwise the property was removed
```

Comment:
In `projectSpecificationHandler()` it will call this function with undefined. When `refPaths` is undefined, the event that is sent from this function will have `refPaths: undefined`? What is the contract with the file-watchers? Does it expect `undefined` or an empty array? cc @jgwest

Reply:
I'd prefer that it always be an empty array, but undefined (or not even specifying the field) is tolerated by the filewatchers.

Reply:
This makes it always an array: https://github.com/eclipse/codewind/pull/1535/commits/52d2690cefcb32ca9e56f0d09a18eee24d612546